src/main/flight/gtune.c

```diff
@@ -0,0 +1,177 @@
+
+#ifdef GTUNE
+
+#include "common/axis.h"
+#include "common/maths.h"
+
+#include "drivers/system.h"
+#include "drivers/sensor.h"
+#include "drivers/accgyro.h"
+
+#include "sensors/sensors.h"
+#include "sensors/acceleration.h"
+
+#include "flight/pid.h"
+#include "flight/imu.h"
+
+#include "config/config.h"
+#include "blackbox/blackbox.h"
+
+#include "io/rc_controls.h"
+
+#include "config/runtime_config.h"
+
+extern uint8_t motorCount;
+
+/*
+ ****************************************************************************
+ ***                    G_Tune                                            ***
+ ****************************************************************************
+	G_Tune Mode
+	This is the multiwii implementation of ZERO-PID Algorithm
+	http://technicaladventure.blogspot.com/2014/06/zero-pids-tuner-for-multirotors.html
+	The algorithm has been originally developed by Mohammad Hefny (mohammad.hefny@gmail.com)
+
+	You may use/modify this algorithm on your own risk, kindly refer to above link in any future distribution.
+ */
+
+/*
+   version 1.0.0: MIN & Maxis & Tuned Band
+   version 1.0.1:
+                a. error is gyro reading not rc - gyro.
+                b. OldError = Error no averaging.
+                c. No Min Maxis BOUNDRY
+    version 1.0.2:
+                a. no boundaries
+                b. I - Factor tune.
+                c. time_skip
+
+   Crashpilot: Reduced to just P tuning in a predefined range - so it is not "zero pid" anymore.
+   Tuning is limited to just work when stick is centered besides that YAW is tuned in non Acro as well.
+   See also:
+   http://diydrones.com/profiles/blogs/zero-pid-tunes-for-multirotors-part-2
+   http://www.multiwii.com/forum/viewtopic.php?f=8&t=5190
+   Gyrosetting 2000DPS
+   GyroScale = (1 / 16,4 ) * RADX(see board.h) = 0,001064225154 digit per rad/s
+
+    pidProfile->gtune_lolimP[ROLL]   = 20; [10..200] Lower limit of ROLL P during G tune.
+    pidProfile->gtune_lolimP[PITCH]  = 20; [10..200] Lower limit of PITCH P during G tune.
+    pidProfile->gtune_lolimP[YAW]    = 20; [10..200] Lower limit of YAW P during G tune.
+    pidProfile->gtune_hilimP[ROLL]   = 70; [0..200]  Higher limit of ROLL P during G tune. 0 Disables tuning for that axisis.
+    pidProfile->gtune_hilimP[PITCH]  = 70; [0..200]  Higher limit of PITCH P during G tune. 0 Disables tuning for that axisis.
+    pidProfile->gtune_hilimP[YAW]    = 70; [0..200]  Higher limit of YAW P during G tune. 0 Disables tuning for that axisis.
+    pidProfile->gtune_pwr            = 0;  [0..10] Strength of adjustment
+*/
+
+static pidProfile_t *pidProfile;
+static int8_t time_skip[3];
+static int16_t OldError[3], result_P64[3];
+static int32_t AvgGyro[3];
+static bool floatPID;
+
+void init_Gtune(pidProfile_t *pidProfileToTune)
+{
+    uint8_t i;
+
+    pidProfile = pidProfileToTune;
+	if (pidProfile->pidController == 2) floatPID = true;                        // LuxFloat is using float values for PID settings

```

Comment:
Would it be cleaner to format the commentns on these conditions like so:

     // LuxFloat is using float values for PID settings
     if (pidProfile->pidController == 2) floatPID = true;